adbms/src/main/scala/de/up/hpi/informationsystems/adbms/relation/TransientRelation.scala

```diff
@@ -18,13 +20,15 @@ private[relation] final class TransientRelation(data: Try[Seq[Record]]) extends
     else
       data.get.head.columns
 
-  /** @inheritdoc*/
-  override def equals(obj: scala.Any): Boolean =
-    obj.isInstanceOf[TransientRelation] &&
-      (hashCode() == obj.asInstanceOf[TransientRelation].hashCode())
+  override def hashCode(): Int = Objects.hashCode(internal_data, isFailure, columns)
+
+  def canEqual(o: Any): Boolean = o.isInstanceOf[TransientRelation]
+
+  override def equals(obj: Any): Boolean = obj match {
+    case that: TransientRelation => that.canEqual(this) && this.hashCode() == that.hashCode()
+    case _ => false
+  }
 
-  override def hashCode(): Int =
-    internal_data.hashCode() + isFailure.hashCode() + columns.hashCode()
```

Comment:
Oh, I was a little fast. Are you not supposed to overwrite `hashCode()` when overwriting `equals()`?

Reply:
i just moved it to the top, see L23